### PR TITLE
build: Initialize Sphinx source

### PR DESCRIPTION
**Pull request by Carlos-Muniz**
Adding necessary files to start a Sphinx based documentation project.
These include the .gitignore, a makefile, requirements.txt, and
requirements.in, which has been mostly copied from the
edx-documentation repo.

### CODE REVIEW

File: docs/make.bat [hunk 1]
```diff
@@ -0,0 +1,35 @@
+@ECHO OFF
```
Comment (feanil) [inline docs/make.bat]: Was this a generated file? It looks like it's to support building the files on Windows? Maybe we don't need it?

File: docs/source/conf.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+#
+# import os
+# import sys
+# sys.path.insert(0, os.path.abspath('.'))
+
+
+# -- Project information -----------------------------------------------------
+
+project = 'tCRIL Onboarding'
+copyright = '2021, tCRIL'
```
Comment (feanil) [inline docs/source/conf.py]: Update this to pull the year dynamically so the copyright stays up-to-date whenever we re-publish.

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1,23 @@
+alabaster==0.7.12
```
Comment (feanil) [inline requirements.txt]: In most openedx projects we use a pattern of having `requirements.in` files that generate the final `requirements.txt` using pip-tools.  I think this is a good pattern and we should continue to use that here.  You can take a look at this is done in for python projects in the [edx-cookiecutter](https://github.com/edx/edx-cookiecutters/tree/master/python-template/%7B%7Bcookiecutter.placeholder_repo_name%7D%7D) repo.  Let me know if you have more questions about this.

**Review by feanil — CHANGES_REQUESTED** [review]

File: docs/make.bat (hunk 1 above)
Comment (Carlos-Muniz) [reply docs/make.bat]: This was a generated file.
I'll dump it since we don't need it.

File: docs/source/conf.py (hunk 2 above)
Comment (Carlos-Muniz) [reply docs/source/conf.py]: The copyright has been updated to be assigned dynamically

**Comment by sarina** [comment]
Looks reasonable to me - I'm satisfied whenever Feanil gives his approval.

File: requirements.txt (hunk 3 above)
Comment (Carlos-Muniz) [reply requirements.txt]: I've added requirements.in, requirements.txt, and a new makefile that can now `make requirements` and `make upgrade`, similarly to the makefile in edx-documentation.

File: docs/source/conf.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+# The theme to use for HTML and HTML Help pages.  See the documentation for
+# a list of builtin themes.
+#
+html_theme = 'alabaster'
+
+# Add any paths that contain custom static files (such as style sheets) here,
+# relative to this directory. They are copied after the builtin static files,
+# so a file named "default.css" will overwrite the builtin "default.css".
+html_static_path = ['_static']
```
Comment (feanil) [inline docs/source/conf.py]: Add a [trailing newline](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline).

File: requirements/requirements.in [hunk 5]
```diff
@@ -0,0 +1,5 @@
+# Core dependencies for documentation builds
+
+sphinx                              # Documentation builder
+edx-sphinx-theme                    # edX theme for the generated documentation
+pip-tools                           # Python requirements file management utilities
```
Comment (feanil) [inline requirements/requirements.in]: Add a [trailing newline](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline).

**Review by feanil — APPROVED** [review]
2 Small things that need to be fixed, then this is good to go.  No need for another round of review before merging.

**Pull request merged**